### ISSUE TITLE
Mismatched types on PathBuf::from()

### ISSUE THREAD

**Issue by Ristarg** [intellij-rust/intellij-rust#2886]
Yet another one of those, eh.

## Environment

* **Intellij-Rust plugin version:** 0.2.0.2106-182
* **Rust toolchain version:** 1.30.0-nightly
* **IDE name and version:** IntelliJ IDEA Community 2018.2.4
* **Operating system:** Windows 10

## Problem description
Plugin incorrectly reports `PathBuf::from(path)` (where `path` is a `String`) as E0308 "mismatched types", suggests the implementation of `From` trait for `Box<Path>`, despite `PathBuf` having an implem of `From` for `String`. Error persists after restarting the IDE.

![image](https://user-images.githubusercontent.com/18281341/46254240-3c538a80-c4b6-11e8-8b38-9278ca889ab9.png)

## Steps to reproduce
Minimal failing example
```rust
fn main() {
    let s = String::new();
    std::path::PathBuf::from(s);
}
```

**Comment by vlad20012** [comment]
Can't reproduce it with the latest nightly rustc and IntelliJ-Rust plugin
